[nicleongyj] IP

nicleongyj opened the pull request:
### Taskmaster : Your personalised task manager 🚀

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Do you ever find yourself overwhelmed by a flood of events and tasks, struggling to remember when and where they all happen? Taskmaster is here to rescue you!

Taskmaster is your personal task manager that liberates your mind, allowing you to focus on what truly matters. It's not just any task manager; it's:
-  💬 Text-based: Simple and intuitive to use.
- 📚 Easy to learn: No steep learning curve.
- ⚡ SUPER FAST: Efficiently manage your tasks without any hassle.

To get started, just follow these simple steps:
1. **Download**: Obtain Taskmaster effortlessly.
2. **Double-Click**: Launch it with ease.
3. **Add Tasks**: Start entering your tasks.
4. **Let it Work for You**: Watch Taskmaster efficiently manage your tasks while you relax.

And it is FREE!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
    public static void main(String[] args) throws DukeException {
        new Taskmaster().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

nicleongyj committed:
Level 0

nicleongyj committed:
level 1

nicleongyj committed:
level 2

nicleongyj committed:
Level 3

nicleongyj committed:
level 4

nicleongyj committed:
level 5

nicleongyj committed:
level 5

nicleongyj committed:
level 6 - delete

nicleongyj committed:
enum

nicleongyj committed:
UI/UX testing

nicleongyj committed:
UI/UX testing

nicleongyj committed:
A-enum

nicleongyj committed:
added save feature

nicleongyj committed:
Merge branch 'branch-Level-7'

nicleongyj committed:
added time/dates for deadlines & events

nicleongyj committed:
added retrieve tasks on date

nicleongyj committed:
Merge branch 'branch-Level-8'

nicleongyj committed:
added more OOP

nicleongyj committed:
added more OOP

nicleongyj committed:
Merge branch 'branch-A-MoreOOP'

nicleongyj committed:
added packages

nicleongyj committed:
Merge branch 'branch-A-Packages'

nicleongyj committed:
Merge branch 'add-gradle-support'

nicleongyj committed:
added gradle

nicleongyj committed:
Merge branch 'branch-A-Gradle'

nicleongyj committed:
added JUnit tests

nicleongyj committed:
Merge branch 'branch-A-Junit'

nicleongyj committed:
added JAR plugins

nicleongyj committed:
fixed some code

nicleongyj committed:
add Java Docs

nicleongyj committed:
Merge branch 'branch-A-JavaDoc'

nicleongyj committed:
modify code to follow coding standards

nicleongyj committed:
fix merge conflict

nicleongyj committed:
fix merge conflict

nicleongyj committed:
add level 9

nicleongyj committed:
fix merge conflict with level-9 branch

seantehds commented on src/main/java/taskmaster/tasks/Task.java:
```diff
@@ -0,0 +1,55 @@
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
```
Pretty nitpicky, but I believe the Coding Standards say you should write this as `setAsDone` instead.

seantehds commented on src/main/java/taskmaster/tasks/TaskList.java:
```diff
@@ -0,0 +1,241 @@
+     * @param date Date that task is happening on.
+     */
+    public void printTasksByDate(String date) {
+        LocalDate dueDate = null;
```
Maybe this should have a modifier? Might want to use `private` if you're able to.

seantehds reviewed (commented):
Overall, well done.

nicleongyj committed:
add JavaFX

nicleongyj committed:
Add FXML skeleton code

nicleongyj committed:
Integrate FXML with Taskmaster logic

nicleongyj committed:
Add CSS styling

nicleongyj committed:
Merge branch 'branch-Level-10'

alientian commented on src/main/java/taskmaster/tasks/Event.java:
```diff
@@ -0,0 +1,76 @@
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
+    /**
+     * Starting time of event in string type.
+     */
+    String startTime;
+    /**
+     * End time of event in string type.
+     */
+    String endTime;
+    /**
+     * Start date of event.
+     */
+    LocalDate startDate;
+    /**
+     * End date of event.
+     */
+    LocalDate endDate;
```
Perhaps you can add access modifier for these variables

alientian commented on src/main/java/taskmaster/tasks/TaskList.java:
```diff
@@ -0,0 +1,246 @@
+     * String representation of task list.
+     * @return String representation of task list.
+     */
+    @Override public String toString() {
```
minor problem, but can write @Override on the line above

alientian commented on src/main/java/taskmaster/tasks/Event.java:
```diff
@@ -0,0 +1,76 @@
+     * @return A string representing the event.
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: "
+                + (startTime == null ? this.startDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")) : startTime)
+                + " to "
+                + (endTime == null ? this.endDate.format(DateTimeFormatter.ofPattern("MMM d yyyy")) : endTime)
+                + ")";
+    }
+}
```
Maybe can compute conditional statements and use String.format(...) to make it easier to read!

alientian reviewed (commented):
Overall looks good! Names are well chosen and follows the coding standard closely.

nicleongyj committed:
update UI source code files

nicleongyj committed:
Add assertions to verify task operations

This commit adds assertions to improve input validation. It ensures user inputs gives a valid commands and verified, and that task indexes is valid and within range, and descriptions is not null

These assertions allows testing of correctness of any assumption that have been made

nicleongyj committed:
Improve code quality for Storage class

nicleongyj committed:
Improve code quality

In this commit, code quality for storage and taskList class is improved. Removed deep nesting, shortened long methods, and restructure some methods to make the happy path prominient

This improvements will contribute to better maintanbility and more readable code

nicleongyj committed:
Merge pull request #1 from nicleongyj/branch-A-Assertions

Add assertions to verify task operations

nicleongyj committed:
Merge branch 'master' into branch-A-CodeQuality

nicleongyj committed:
Refactor addTask method

nicleongyj committed:
Add duplicate checker to detect and prevent duplicate tasks

-Implemented a feature to detect duplicate descriptions for todo tasks

-Implemented checking feature to check duplicate description and timings for deadlines and events

nicleongyj committed:
Merge branch 'branch-BCD-Extension'

nicleongyj committed:
Merge branch 'master' into branch-A-CodeQuality

nicleongyj committed:
Merge pull request #3 from nicleongyj/branch-A-CodeQuality

Branch a code quality

nicleongyj committed:
Add javadocs to methods and refactored AddTask method

nicleongyj committed:
Fix storage functionality for JAR files

nicleongyj committed:
Merge branch 'branch-FixStorage'

nicleongyj committed:
Finalise code and improve code quality

nicleongyj committed:
Update README.md

Added in feature list and usages

nicleongyj committed:
Update README.md

nicleongyj committed:
add Ui picture to docs

nicleongyj committed:
Merge branch 'master' of https://github.com/nicleongyj/ip

nicleongyj committed:
Fix styling for javadocs

nicleongyj committed:
Update README.md

nicleongyj committed:
Update README.md
